go/test/endtoend/multiorch/dead_primary_recovery_test.go

```diff
@@ -115,3 +115,154 @@ func TestDeadPrimaryRecovery(t *testing.T) {
+// primary's multipooler process is down but Postgres is still running and replicas are
+// still connected to it.
+//
+// This tests the design decision that when only the pooler process crashes (not Postgres),
+// the operator should restart the pooler rather than triggering an automatic failover.
+// This avoids unnecessary failovers for OOM or process crashes that don't affect the database.
+func TestPoolerDownNoFailover(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping end-to-end pooler down test (short mode)")
+	}
+	if utils.ShouldSkipRealPostgres() {
+		t.Skip("Skipping end-to-end pooler down test (short mode or no postgres binaries)")
+	}
+
+	_, err := exec.LookPath("etcd")
+	require.NoError(t, err, "etcd binary must be available in PATH")
+
+	// Setup test environment
+	env := setupMultiOrchTestEnv(t, testEnvConfig{
+		tempDirPrefix:    "poolerdown*",
+		cellName:         "test-cell",
+		database:         "postgres",
+		shardID:          "test-shard-poolerdown",
+		tableGroup:       "test",
+		durabilityPolicy: "ANY_2",
+		stanzaName:       "poolerdown-test",
+	})
+
+	// Create 3 nodes
+	nodes := env.createNodes(3)
+	t.Logf("Created 3 empty nodes")
+
+	// Setup pgbackrest
+	env.setupPgBackRest()
+
+	// Register all nodes in topology
+	env.registerNodes()
+
+	// Start multiorch
+	env.startMultiOrch()
+
+	// Wait for initial bootstrap
+	t.Logf("Waiting for initial bootstrap...")
+	primaryNode := waitForShardPrimary(t, nodes, 60*time.Second)
+	require.NotNil(t, primaryNode)
+	t.Logf("Initial primary: %s", primaryNode.name)
+
+	// Wait for standbys to complete initialization
+	t.Logf("Waiting for standbys to complete initialization...")
+	waitForStandbysInitialized(t, nodes, primaryNode.name, len(nodes)-1, 60*time.Second)
+
+	// Verify standbys are replicating from the primary
+	t.Logf("Verifying standbys are replicating from primary...")
+	for _, node := range nodes {
+		if node.name == primaryNode.name {
+			continue
+		}
+		status := checkInitializationStatus(t, node)
+		require.NotNil(t, status.ReplicationStatus, "Standby %s should have replication status", node.name)
+		require.NotNil(t, status.ReplicationStatus.PrimaryConnInfo, "Standby %s should have PrimaryConnInfo", node.name)
+		t.Logf("Standby %s is replicating from %s:%d",
+			node.name,
+			status.ReplicationStatus.PrimaryConnInfo.Host,
+			status.ReplicationStatus.PrimaryConnInfo.Port)
+	}
+
+	// Kill the multipooler process on the primary (but leave Postgres running)
+	t.Logf("Killing multipooler on primary node %s (postgres stays running)", primaryNode.name)
+	killMultipooler(t, primaryNode)
+
+	// Wait for multiorch to detect the pooler is down and run several recovery cycles.
+	// Tests configure pooler-health-check-interval and recovery-cycle-interval to 500ms,
+	// so 3 seconds allows ~6 cycles which is plenty to detect and process the state.
+	t.Logf("Waiting for multiorch to process the pooler down state...")
+	time.Sleep(3 * time.Second)
```

Comment:
Not a blocking comment, but for faster tests in the future maybe we should have a synchronous HTTP request or something to force multiorch to run recovery on-demand during endtoend tests instead of just waiting